language/move-vm/types/src/values/values_impl.rs

```diff
@@ -1423,9 +1423,9 @@ pub mod vector {
             (Type::U128, Container::U128(v)) => v.len(),
             (Type::Bool, Container::Bool(v)) => v.len(),
 
-            (Type::Struct(_), Container::General(v)) | (Type::Address, Container::General(v)) => {
-                v.len()
-            }
+            (Type::Struct(_), Container::General(v))
+            | (Type::Vector(_), Container::General(v))
+            | (Type::Address, Container::General(v)) => v.len(),
 
             (tag, v) => err_vector_elem_ty_mismatch!(tag, v),
```

Comment:
Could you add exhaustive pattern matching here (and elsewhere in the code you had to change)? I'm guessing this bug was introduced when `Type::Vector`was introduced and the compiler didn't warn us that this code needed to be changed. Exhaustive pattern matching would fix that.

Reply:
That's a really good point. A full expansion may be a little bit too verbose, but let me see if I can change the last pattern from "(_, _)" to "(Type::U8, _), (Type::U64, _), ...". This should be sufficient as the `Type` is always more refined than the enum tags the `Container` carries.